### PR TITLE
Install headers to include/${PROJECT_NAME}

### PR DESCRIPTION
**Pull request by sloretz**
Part of ros2/ros2#1150 - this installs headers to a unique directory to avoid include directory search order issues when overriding packages.

The patch is taken from https://github.com/ros/console_bridge/pull/94

### CODE REVIEW

**Review by mjcarroll — DISMISSED** [review]

**Comment by sloretz** [comment]
CI (build: `--packages-above-and-dependencies console_bridge_vendor` test: `--packages-select console_bridge_vendor`)
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/16291/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/10911/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/16658/)

**Comment by sloretz** [comment]
CI re-run with new commit

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/16294/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/10914/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/16661/)

**Comment by sloretz** [comment]
The new commit makes the console_bridge_vendor tests get the include directories and libraries from the built `console_bridge-config.cmake` file instead of hard coding it. Hopefully this fixes the windows failure.

**Comment by sloretz** [comment]
Turns out mixing `find_package()` and `ExternalProject_add` doesn't work. I read about alternatives and decided updating the hardcoded include directory is the way to go.

**Comment by sloretz** [comment]
CI Re-run again
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/16295/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/10915/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/16662/)

**Review by clalancette — COMMENTED** [review]
So, this whole thing is pretty unfortunate.  Ubuntu 22.04 ships with console_bridge 1.0.1 (the latest tag), but does not include this commit (since we haven't merged and released https://github.com/ros/console_bridge/pull/94 yet).  Since I don't think that we are going to get 1.0.2 (or whatever we call the next version) into Jammy, I think we are going to have to build this vendor package there as well.

With that said, then, I don't think we should merge this as-is.  I think we should get https://github.com/ros/console_bridge/pull/94 merged and a new release done, and then we should change this PR to require 1.0.2 or greater to make sure we are building it appropriately.  @sloretz Does that sound reasonable?

**Comment by sloretz** [comment]
> So, this whole thing is pretty unfortunate. Ubuntu 22.04 ships with console_bridge 1.0.1 (the latest tag), but does not include this commit (since we haven't merged and released https://github.com/ros/console_bridge/pull/94 yet). Since I don't think that we are going to get 1.0.2 (or whatever we call the next version) into Jammy, I think we are going to have to build this vendor package there as well.

Ubuntu 22.04 ships with console_bridge? Include directories from system packages don't matter since they will be `/usr/include` or something like that, not `/opt/ros/.../include`.

> I think we should get https://github.com/ros/console_bridge/pull/94 merged and a new release done, and then we should change this PR to require 1.0.2 or greater to make sure we are building it appropriately. @sloretz Does that sound reasonable?

I don't think it's necessary. If ros/console_bridge#94 is the only change, then there's no reason to require 1.0.2 over 1.0.1 on platforms where 1.0.1 is available from the system.

**Comment by sloretz** [comment]
@clalancette Mind having a second look?

Now that `console_bridge` has a 1.0.2 tag I updated this PR to look for at least version 1.0.1, but if that's not found then to build 1.0.2. That allows using the Jammy Package because the include directory search order problem is only relevant if the headers are in a ROS workspace, and allows overriding this package when the vendored `console_bridge` version is built.

File: CMakeLists.txt [hunk 1]
```diff
@@ -64,10 +64,14 @@ macro(build_console_bridge)
...
-    URL https://github.com/ros/console_bridge/archive/1.0.1.tar.gz
-    URL_MD5 9f5351994c7690b37f88e9f70e478061
+  # Download and build console_bridge
+  externalproject_add(console_bridge-1.0.2
+    GIT_REPOSITORY https://github.com/ros/console_bridge.git
+    GIT_TAG 0828d846f2d4940b4e2b5075c6c724991d0cd308 # 1.0.2
+    GIT_CONFIG advice.detachedHead=false
+    # Suppress git update due to https://gitlab.kitware.com/cmake/cmake/-/issues/16419
+    # See https://github.com/ament/uncrustify_vendor/pull/22 for details
+    UPDATE_COMMAND ""
```
Comment (clalancette) [inline CMakeLists.txt]: Now that we have a tagged upstream version, and don't need to patch anymore, can we switch back to `URL`/`URL_MD5`?

**Review by clalancette — COMMENTED** [review]
One question inline, otherwise looks good.

File: CMakeLists.txt (hunk 1 above)
Comment (sloretz) [reply CMakeLists.txt]: Back to using `URL` and a hash in b654310

**Comment by sloretz** [comment]
CI (build: ` --packages-above-and-dependencies console_bridge_vendor` test: `--packages-select console_bridge_vendor`)
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/16314/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/10934/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/16684/)

**Review by clalancette — APPROVED** [review]

**Pull request merged**